don't crash on a single PVC resizing failure

anas-aso opened the pull request:
The controller shouldn't crash if a single PVC resize failure happens.

I run the controller with the flag `--no-annotation-check`, so it might be the case that someone sets the controller annotations on a PVC that is not resizable (StorageClass does not allow volume expansion). Right now this leads to stopping the processing for all PVCs.

I think it is safe to fail that single PVC and keep processing the others. Until we have metrics, relying on logged errors should be enough. WDYT ?

toshipp commented:
We tried to fix the same issue #24, but the fix makes it harder to notice error occurrences.
We conclude that exporting metrics introduced by #25 is needed.

anas-aso commented:
@toshipp Thanks for your quick reply. Sorry I wasn't clear about the motivation of my PR. I have updated the PR tile and description to reflect my point, and also changed the code. Let me know WDYT.

toshipp commented:
I understand your situation, this is acceptable, IMO.
@llamerada-jp How do you think about it?

toshipp commented on runners/pvc_autoresizer.go:
```diff
@@ -118,7 +118,7 @@ func (w *pvcAutoresizer) reconcile(ctx context.Context) error {
 			err = w.resize(ctx, &pvc, vsMap[namespacedName])
 			if err != nil {
-				return err
+				w.log.WithValues("namespace", pvc.Namespace, "name", pvc.Name).Error(err, "failed to resive PVC")
```
I assume "resive" is typo of resize, isn't it?

toshipp commented:
Would you squash commits? I don't want to import revert commits.

llamerada-jp commented:
@anas-aso @toshipp 
Sorry for the late reply.
I realized that the program before the fix would not expand the PVC that caused the error as well as other PVCs. It would be better to fix it by the current PR.

toshipp reviewed (approved):
LGTM

llamerada-jp reviewed (approved):
LGTM

anas-aso commented:
@llamerada-jp @toshipp any idea what is your release cycle ? It is not documented anywhere in the repo, and it seems that you don't build a docker image for the main branch (with `latest` tag). There are some nice fixes since the last release, and I don't want to build my own docker image for them 😅

satoru-takeuchi commented:
@anas-aso We don't have strict release cycle. We'll release new version next week.
